tests/JoinArrayTest.php

```diff
@@ -60,7 +60,7 @@ public function testDirection2()
     public function testJoinSaving1()
     {
         $a = ['user' => [], 'contact' => []];
-        $db = new Persistence\Array_($a);
+        $db = new Persistence\Array_(new Persistence\Array_\ArrayByRef($a));
```

Comment:
Shouldn't Persistence/Array_ automatically use ArrayByRef by default somehow? That way we wouldn't need to change this line in our projects. Or ArrayByRef would be one level up in namespace tree Persistence/ArrayByRef.

Reply:
No, I am think now more about removing the ArrayByRef totally - what is the usecase of allowing arrays by reference?

You can always hack it by assigning `data` directly in array persistence...

Reply:
This looks weird. ArrayByRef shouldn't probably be this deep in the hierarchy. It's also a proxy class and some basic functionality may take a significant performance hit because of this.